main_test.go

```diff
@@ -44,51 +89,230 @@ webhook:
 		t.Fatal(err)
 	}
 
+	disableCertOverWriteConfig := filepath.Join(tmpDir, "disable-cert-overwrite.yaml")
+	if err := os.WriteFile(disableCertOverWriteConfig, []byte(`
+apiVersion: config.kueue.x-k8s.io/v1alpha2
+kind: Configuration
+namespace: kueue-system
+health:
+  healthProbeBindAddress: :8081
+metrics:
+  bindAddress: :8080
+leaderElection:
+  leaderElect: true
+  resourceName: c1f6bfd2.kueue.x-k8s.io
+webhook:
+  port: 9443
+internalCertManagement:
+  enable: false
+`), os.FileMode(0600)); err != nil {
+		t.Fatal(err)
+	}
+
+	defaultControlOptions := ctrl.Options{
+		Port:                   9443,
+		HealthProbeBindAddress: defaultHealthProbeAddress,
+		MetricsBindAddress:     defaultMetricsAddress,
+		LeaderElectionID:       defaultLeaderElectionID,
+		LeaderElection:         true,
+	}
+
+	enableDefaultInternalCertManagement := &configv1alpha2.InternalCertManagement{
+		Enable:             pointer.Bool(true),
+		WebhookServiceName: pointer.String(defaultServiceName),
+		WebhookSecretName:  pointer.String(defaultSecretName),
+	}
+
+	cmpOpts := []cmp.Option{
+		cmpopts.IgnoreFields(ctrl.Options{}, "Scheme",
+			"makeBroadcaster",
+			"Logger",
+			"newRecorderProvider",
+			"newResourceLock",
+			"newMetricsListener",
+			"newHealthProbeListener",
+		),
+	}
+
 	testcases := []struct {
-		name                             string
-		configFile                       string
-		wantPort                         int
-		wantEnableInternalCertManagement bool
+		name              string
+		configFile        string
+		wantConfiguration configv1alpha2.Configuration
+		wantOptions       ctrl.Options
 	}{
 		{
-			name:                             "default config",
-			configFile:                       "",
-			wantPort:                         9443,
-			wantEnableInternalCertManagement: true,
+			name:       "default config",
+			configFile: "",
+			wantConfiguration: configv1alpha2.Configuration{
+				Namespace: pointer.String(defaultNamespace),
+				InternalCertManagement: &configv1alpha2.InternalCertManagement{
+					Enable:             pointer.Bool(true),
+					WebhookServiceName: pointer.String(defaultServiceName),
+					WebhookSecretName:  pointer.String(defaultSecretName),
+				},
+			},
+			wantOptions: ctrl.Options{
+				Port:                   9443,
+				HealthProbeBindAddress: defaultHealthProbeAddress,
+				MetricsBindAddress:     defaultMetricsAddress,
+				LeaderElectionID:       defaultLeaderElectionID,
+				LeaderElection:         false,
+			},
+		},
+		{
+			name:       "namespace overwrite config",
+			configFile: namespaceOverWriteConfig,
+			wantConfiguration: configv1alpha2.Configuration{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: configv1alpha2.GroupVersion.String(),
+					Kind:       "Configuration",
+				},
+				Namespace:                  pointer.String("kueue-tenant-a"),
+				ManageJobsWithoutQueueName: false,
+				ControllerManagerConfigurationSpec: cfg.ControllerManagerConfigurationSpec{
```

Comment:
We can ignore the field when calling diff for this is already checked in `wantOptions`. Code like
```
if diff := cmp.Diff(tc.wantConfiguration, config, cmpopts.IgnoreFields(configv1alpha2.Configuration{}, "ControllerManagerConfigurationSpec")); diff != "" {
}
```

Reply:
It makes sense.

Reply:
Done.